src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -147,10 +147,15 @@ def test_delete_failure_not_found(resource_client, current_resource_model):
 
 
 def test_input_equals_output(resource_client, input_model, output_model):
+
     pruned_input_model = prune_properties_from_model(
         input_model.copy(), resource_client.write_only_paths
     )
 
+    pruned_input_model = prune_properties_from_model(
+        pruned_input_model, resource_client.read_only_paths
+    )
+
```

Comment:
why does the input model have readOnly property? ReadOnly properties should be set by the service when creating the resources

Reply:
Good question. The input model shouldn't have a readOnly property, but for the update contract tests, the output model for the create is used as the input model for the subsequent update. See [handler_update.py#L47-L53](https://github.com/aws-cloudformation/cloudformation-cli/blob/master/src/rpdk/core/contract/suite/handler_update.py#L47-L53), [handler_update.py#L23-L28](https://github.com/aws-cloudformation/cloudformation-cli/blob/master/src/rpdk/core/contract/suite/handler_update.py#L23-L28), and [resource_client.py#L255-L260](https://github.com/aws-cloudformation/cloudformation-cli/blob/009e0965d32f51ce94f84af2c5a33698273bc515/src/rpdk/core/contract/resource_client.py#L255-L260).

Reply:
Do you think the pruning should happen in `resource_client.generate_update_example` instead?

Reply:
Checked with the team, read_only properties should be updatable using update handler. I can go ahead approve this PR, could you please move the change to update?

Reply:
Yup, sounds good. I will move these lines into `generate_update_example`.

Reply:
Actually, after taking another look, `generate_update_example` is used for more than the update_read_success and update_list_success contract tests. Instead, I'll move this into the `pytest.fixture` - `updated_resource`, which is only used by the two affected tests.